sdk/src/wallet/types/address.rs

```diff
@@ -1,18 +1,34 @@
 // Copyright 2021 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
+use core::ops;
+
+use derive_more::{AsRef, From};
 use getset::{Getters, Setters};
 use serde::{Deserialize, Serialize};
 
-use crate::types::block::{address::Bech32Address, output::OutputId};
+use crate::{
+    types::block::{address::Bech32Address, output::OutputId},
+    wallet::OutputWithExtendedMetadata,
+};
 
-/// An account address with unspent output_ids for unspent outputs.
-#[derive(Debug, Getters, Setters, Clone, Serialize, Deserialize, Eq, PartialEq)]
+#[derive(Debug, Getters, Setters, Clone, Serialize, Deserialize, Eq, PartialEq, derive_more::Deref)]
+#[serde(rename_all = "camelCase")]
+#[getset(get = "pub")]
+pub(crate) struct AddressWithUnspentOutputIds {
```

Comment:
Why add this and not just use AddressWithUnspentOutputs where the unspent_outputs are empty as long as there are none? There are some places where you do conversions between the two types that don't seem necessary if it would be just a single type

Reply:
it's actually making things less annoying as I don't need to destructure in a few cases